02_bubble/pfannerj/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var mainout io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintf(mainout, "Bubble sort starting...\n")
+	fmt.Fprintln(mainout, bubble([]int{3, 2, 1, 5}))
+	fmt.Fprintf(mainout, "Bubble sort completed...\n")
+	fmt.Fprintf(mainout, "Insertion sort starting...\n")
+	fmt.Fprintln(mainout, insertion([]int{3, 2, 5, 1}))
+	fmt.Fprintf(mainout, "Insertion sort completed...\n")
+}
+
+func bubble(s []int) []int {
+	sCopy := make([]int, len(s))
```

Comment:
Bubble sort can be done without creating another copy of the slice, and all operations can be done on the input slice. This may impact performance with large input.

Reply:
Creating a copy is in the requirements... Refer to the readme - "Write a function that returns a sorted **copy** of `int` slice `s` using bubble sort..."

Reply:
I will try to clarify that on the slack channel, but if you pass the slice as an argument into a function, you won't get immutability.

Reply:
FYI.  It was specifically mentioned by Dan during the session - in my notes I wrote this "Needs to return a copy of the int slice ’s’ (don’t modify s)"